torch/csrc/jit/passes/peephole.cpp

```diff
@@ -46,7 +47,43 @@ void PeepholeOptimize(Block * block) {
             }
           }
         }
-        break;
+      } break;
+      // Fuse mm + add into addmm
+      case aten::add: {
+        // Must have two inputs
+        if (n->inputs().size() != 2) {
+          continue;
+        }
+        // Alpha parameter must be 1.0
+        auto alpha = at::Scalar(it->t(attr::alpha));
+        if (alpha.to<double>() != 1.0) {
+          continue;
+        }
+
+        auto input_node = n->input(1)->node();
+        // Input must be an mm node
+        if (input_node->kind() != aten::mm) {
+          continue;
+        }
+
+        WithInsertPoint guard(n);
+
+        SymbolicVariable mat(n->input(0));
+        SymbolicVariable mat1(input_node->input(0));
+        SymbolicVariable mat2(input_node->input(1));
+        SymbolicVariable addmm_value = mat.addmm(mat1, mat2);
+
+        n->output()->replaceAllUsesWith(addmm_value);
+        it.destroyCurrent();
+
+        if (input_node->output()->uses().size() == 0) {
```

Comment:
Can we let DCE fix up this left over node? I feel like a lot of the peephole passes will want to have that cleanup anyway, and the check for `*it == input_node` is pretty obscure.